ui/src/VoxettaRecordButton.js

```diff
@@ -16,39 +16,85 @@ limitations under the License. */
 import {AudioRecorder} from './AudioRecorder';
 import {LitElement, html, css} from 'lit-element';
 
+/**
+ * Component responsible for recording and uploading audio files. 
+ */
 export class VoxettaRecordButton extends LitElement {
     static get properties() {
         return {
             isRecording: {type: Boolean},
             audioRecorder: {type: Object},
         };
     }
+
     constructor() {
         super();
         this.isRecording = false;
         this.audioRecorder = new AudioRecorder();
     }
-    render() {  
+
+    render() {
+        this.getUploadUrl();  
         return html`
             <button @click=${this.recordHandler}>Record Voice</button>
             <audio id="utterance" controls src="" style="display: none"></audio>
         `;
     }
 
-     async recordHandler(e){
+    /**
+     * If the user is not currently recording, begin recording using the Microphone component. Otherwise,
+     * stop recording and save and display the just-recorded audio file.
+     */
+    async recordHandler(){
         if (!this.isRecording) {
             this.isRecording = true;
             this.audioRecorder.startRecording();
         } else {
-            const audioSave = this.shadowRoot.getElementById("utterance");
             this.isRecording = false;
-            const recordingUrl = await this.audioRecorder.stopRecording();
-            if(recordingUrl != null){
-                audioSave.src = recordingUrl;
+            const audioSave = this.shadowRoot.getElementById("utterance");
```

Comment:
In a future change, you'll probably turn this utterance into its own component called something like <vox-utterance-list> and pass in the src and display properties in the render function instead of searching for it like this.

This should probably be two sibling components, since the utterance list will be positioned differently on desktop. It's easier to manage positioning of siblings instead of children of children, via css grid.

```
<parent>
  <vox-button/>
  <vox-utt-list/>
</parent>
```

Reply:
This makes sense! I was planning on leaving it for now because I wasn't sure if it was to be incorporated into the MVP UI or not--will bring up in STEP Cafe or Pod Sync today